src/ape/contracts/base.py

```diff
@@ -231,6 +236,13 @@ class ContractInstance(AddressAPI):
     _converter: "ConversionManager"
     _contract_type: ContractType
 
+    def __post_init__(self):
+        if not self.is_contract:
```

Comment:
K that makes sense but currently `ContractCall` and `ContractTransaction` do not have access to the `AddressAPI`. They get initialized with `AddressType`. We could instead pass along the `ContractInstance` in place of the `AddressType` as a means of refactor to handle this situation. Else, we can make the exception happen in `ContractInstance.__getattr__()`, which may work but is a little strange I think because the error would raise without actually calling the method.

Reply:
Ah, I see what you mean now.

How bad would it be to pass the full `ContractInstance` object as a backwards ref?

Reply:
I just did and it works okay, I am just keeping the type `AddressAPI` since all we need is the `is_contract` method